calculator.go

```diff
@@ -1,13 +1,97 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+)
+
+// Add takes a list of numbers and returns the result of adding them together.
+func Add(a, b float64, nums ...float64) float64 {
+	result := a + b
+	for _, n := range nums {
+		result += n
+	}
+	return result
+}
+
+// Subtract takes a list of numbers and returns the result of subtracting them from each other.
+func Subtract(a, b float64, nums ...float64) float64 {
+	result := a - b
+	for _, n := range nums {
+		result -= n
+	}
+	return result
+}
+
+// Multiply takes a list of numbers and multiplies them.
+func Multiply(a, b float64, nums ...float64) float64 {
+	result := a * b
+	for _, n := range nums {
+		result *= n
+	}
+	return result
+}
+
+// Divide takes a list of numbers and divides them.
+// Error is returned when dividing by 0.
+func Divide(a, b float64, nums ...float64) (float64, error) {
+	err := errors.New("cannot divide by zero")
+	if a == 0 || b == 0 {
```

Comment:
But if `a` is zero, that's not an error!

Reply:
addressed in https://github.com/micuffaro/ftl-fundamentals/pull/1/commits/7f0584f8ddefaae0f23bb0d586254ebb26619421